data/metadata/metadata.go

```diff
@@ -30,20 +36,37 @@ func (m *Metadata) EqualsTo(b *Metadata) bool {
 		return false
 	}
 
-	return reflect.DeepEqual(m.GetTags(), b.GetTags())
+	return reflect.DeepEqual(m.GetMetadataMap(), b.GetMetadataMap())
+}
+
+// AddMetadata adds a generic metadata to the entity
+func (m *Metadata) AddMetadata(key string, value interface{}) {
+	m.Metadata[key] = value
+}
+
+// GetMetadata gets a specific metadata from the metadata section of an entity
+func (m *Metadata) GetMetadata(key string) interface{} {
+	return m.Metadata[key]
 }
 
 // AddTag adds a tag to the entity
 func (m *Metadata) AddTag(key string, value interface{}) {
-	m.Tags[key] = value
+	m.Metadata[applyTagsPrefix(key)] = value
 }
 
 // GetTag gets a specific tag from the metadata section of an entity
 func (m *Metadata) GetTag(key string) interface{} {
-	return m.Tags[key]
+	return m.Metadata[applyTagsPrefix(key)]
+}
+
+// GetMetadataMap gets all the tags added to the metadata section of an entity
+func (m *Metadata) GetMetadataMap() Map {
```

Comment:
not sure if this is useful since the Metadata attribute of the struct is already exported..

Reply:
yeah, it's not very useful